opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/presenter/BaseIssueReferralPresenter.java

```diff
@@ -0,0 +1,162 @@
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+        return null;
+
+    }
+
+    @Override
+    public String getMainTable() {
+        return "ec_family_member";
+    }
+
+    @Override
+    public void fillClientData(@Nullable MemberObject memberObject) {
+        if (memberObject != null && getView() != null) {
+            getView().setProfileViewWithData();
+        }
+    }
+
+    @Override
+    public void initializeMemberObject(MemberObject memberObject) {
+        this.MEMBER_OBJECT = memberObject;
+    }
+
+
+    @Override
+    public void saveForm(String jsonString) {
+        try {
+            interactor.saveRegistration(jsonString, this);
+        } catch (Exception e) {
+            Timber.e(Log.getStackTraceString(e));
+        }
+    }
+
+
+    @Override
+    public void onUniqueIdFetched(Triple<String, String, String> triple, String entityId) {
+//        implement
+    }
+
+    @Override
+    public void onNoUniqueId() {
+//        implement
+    }
+
+    @Override
+    public void onRegistrationSaved() {
+//        implement
+    }
+
+    @Override
+    public void registerViewConfigurations(List<String> list) {
+//        implement
+    }
+
+    @Override
+    public void unregisterViewConfiguration(List<String> list) {
+//        implement
+    }
+
+    @Override
+    public void onDestroy(boolean b) {
+//        implement
+    }
+
+    @Override
+    public void updateInitials() {
+//        implement
+    }
+
+
+    private BaseIssueReferralContract.View getView() {
+        if (viewReference != null)
+            return viewReference.get();
+        else
+            return null;
+    }
+
+
+    @Override
+    public boolean validateValues(MemberObject memberObject) {
+        String message = "Error: ";
+
+        if (TextUtils.isEmpty(memberObject.getChwReferralHf()) || memberObject.getChwReferralHf() == null) {
+            try {
+                message = getView().getCurrentContext().getResources().getString(R.string.missing_facility);
+                makeToast(message);
+            } catch (Exception e) {
+                Timber.e(e);
+            }
+            return false;
+
+        } else if (TextUtils.isEmpty(memberObject.getChwReferralService()) || memberObject.getChwReferralService() == null) {
+            try {
+                message = getView().getCurrentContext().getResources().getString(R.string.missing_services);
+                makeToast(message);
+            } catch (Exception e) {
+                Timber.e(e);
+            }
+            return false;
+
+        } else
+            return true;
+    }
+
+    private void makeToast(String message) {
+        try {
+            Toast.makeText(ReferralLibrary.getInstance().context().applicationContext(),
+                    message,
+                    Toast.LENGTH_LONG).show();
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+    }
```

Comment:
You can make use of the utility method ```Utils.showToast();``` available in the **opensrp-client-core** module and delete this. For Short toast messages you could use ```Utils.showShortToast();``` Both have different params.

Reply:
Noted